python/dgl/distributed/optim/pytorch/sparse_optim.py

```diff
@@ -323,7 +327,11 @@ def step(self):
                                 idx_j = idx_i[mask]
                                 grad_j = grad_i[mask]
                                 idx_split_size.append(
-                                    th.tensor([idx_j.shape[0]], dtype=th.int64)
+                                    th.tensor(
+                                        [idx_j.shape[0]],
+                                        dtype=th.int64,
+                                        device=device,
```

Comment:
what if multiple GPUs are being used during train, do we need to specify device id like `cuda:0/1/2/3` here?

Reply:
I think you are right. Changed device to   `device = th.device(f"cuda:{self._rank}")`